runtime/caml/runtime_events.h

```diff
@@ -226,7 +226,7 @@ CAMLextern char_os* caml_runtime_events_current_location();
 void caml_ev_begin(ev_runtime_phase phase);
 void caml_ev_end(ev_runtime_phase phase);
 void caml_ev_counter(ev_runtime_counter counter, uint64_t val);
-void caml_ev_lifecycle(ev_lifecycle lifecycle, int64_t data);
+CAMLextern void caml_ev_lifecycle(ev_lifecycle lifecycle, int64_t data);
```

Comment:
it's a bit weird, why is only one function CAMLextern and not the others?

Reply:
To answer it in two parts: it is marked `CAMLextern` because it is used (via the `CAML_EV_LIFECYCLE` macro) in `otherlibs/unix/fork.c` and has to be marked `__declspec(dllimport)` in order for FlexDLL to be able to relocate `dllunix.so`. It affects Cygwin only (and not also mingw-w64 in 5.0) just because mingw-w64 implementation doesn't compile `fork.c`.

I hadn't marked the others because it's all tied into the work I previously did trying to get towards using fvisiblity and that - it turns out - got tied into an obscure little bit of the relocatable work (full explanation is a bit big for the margins of this GitHub comment...). So I'd left it as being the minimal fix needed to get Cygwin64 back up for now! I don't mind speculatively going through the headers and adding `CAMLextern`; it's just not strictly related to the PR subject of restoring the Cygwin port.

Reply:
I would prefer if either all those `caml_ev_` functions had the same status (so yes, adding CAMLextern to all of them would be fine), or if there was a comment in the code to explain the inconsistency (and I think this is not as good as being consistent in the first place if there is no obvious downside to consistency).

Reply:
I've added a comment - adding `__declspec(dllimport)` is not a zero-cost operation, although the performance impact depends on how hot the function calls are (cf. the Windows TLS PRs). I have previously agreed that `CAMLextern` is an unfortunate macro name (especially as function declarations don't require the `extern` keyword anyway) - `CAML_PUBLIC_API` or `CAML_NEEDS_TO_BE_VISIBLE_FROM_SHARED_OBJECTS` removes the link from C's `extern` keyword!

Reply:
In the present case, it is unclear to me why we would want to make just one function visible here and not the other (I would say they are all equally useful to users of our public api), so even if there is a cost the argument could be made that they should all be exported, or someone will be limited later and have to complain to ask us to export more. But if you think that minimal export is the right default, and there is a comment to explain that choice, then it is fine with me.